File: spark-connector/spark-connector/src/main/scala/org/apache/spark/sql/iceberg/extensions/GravitinoIcebergSparkSessionExtensions.scala
/*
 *  Copyright 2024 Datastrato Pvt Ltd.
 *  This software is licensed under the Apache License version 2.
 */

package org.apache.spark.sql.iceberg.extensions

import org.apache.spark.sql.SparkSessionExtensions
import org.apache.spark.sql.iceberg.catalyst.parser.GravitinoIcebergSparkSqlExtensionsParser

class GravitinoIcebergSparkSessionExtensions extends (SparkSessionExtensions => Unit) {

  override def apply(extensions: SparkSessionExtensions): Unit = {
      // parser extensions
      extensions.injectParser { case (_, parser) => new GravitinoIcebergSparkSqlExtensionsParser(parser) }
  }
}

File: spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/extensions/GravitinoIcebergSparkSessionExtensions.java
/*
 * Copyright 2024 Datastrato Pvt Ltd.
 * This software is licensed under the Apache License version 2.
 */
package com.datastrato.gravitino.spark.connector.iceberg.extensions;

import org.apache.spark.sql.SparkSessionExtensions;
import scala.Function1;

public class GravitinoIcebergSparkSessionExtensions
    implements Function1<SparkSessionExtensions, Void> {

  @Override
  public Void apply(SparkSessionExtensions extensions) {

    // planner extensions
    extensions.injectPlannerStrategy(IcebergExtendedDataSourceV2Strategy::new);

    // There must be a return value, and Void only supports returning null, not other types.
    return null;
  }
}
